### PR TITLE
Update machine image for GPU CI

### PR DESCRIPTION
**Pull request by ivirshup**
- [ ] Closes #
- [ ] Tests added
- [ ] Release note added (or unnecessary)

Trying a new AMI for GPU CI

### CODE REVIEW

File: .cirun.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ runners:
   - name: aws-gpu-runner
     cloud: aws
     instance_type: g4dn.xlarge
-    machine_image: ami-0678adbdcb4c3a662
+    machine_image: ami-0139a05e194cbd9a0
```
Comment (aktech) [inline .cirun.yml:5]: Note: This failed here: https://github.com/scverse/anndata/pull/1084/checks?check_run_id=15487580477
Is this AMI in `eu-west-1` (that's default region)? If not, then you need to add `region: <region-name>` in here.

Comment (ivirshup) [reply .cirun.yml]: I believe this didn't work because I didn't wait for the AMI to build to stop the instance, so the AMI failed to build.

Comment (ivirshup) [reply .cirun.yml]: How would I tell what region it's associated with? I added eu-north-1 for the new one, but it's not super clear from the UI

Comment (aktech) [reply .cirun.yml]: It's visible on the top right in the aws console, you already got it anyway. :)

**Pull request merged**